redis/src/bloom.rs

```diff
@@ -0,0 +1,310 @@
+                        Value::Int(i) => Some(*i),
+                        _ => None,
+                    }),
+                };
+                Ok(resp)
+            }
+            Value::Map(map) => {
+                let resp = SingleInfoResponse {
+                    value: map.first().and_then(|(_, v)| match v {
+                        Value::Int(i) => Some(*i),
+                        _ => None,
+                    }),
+                };
+                Ok(resp)
+            }
+
+            _ => Err(not_convertible_error!(
+                v,
+                "expected vector (resp2) or map (resp3)"
+            )),
+        }
+    }
+}
+
+/// The result of bf_scan_dump
+/// See the [Redis documentation](https://redis.io/docs/latest/commands/bf.scandump/) for more information.
+#[derive(Debug, Clone)]
+pub struct DumpResult {
+    /// The iterator id to be used in the next call to `bf_scan_dump`
+    pub iterator: i64,
+    /// The chunked data
+    pub data: Vec<u8>,
+}
+
+impl FromRedisValue for DumpResult {
+    fn from_redis_value(v: &Value) -> RedisResult<Self> {
+        match v {
+            Value::Array(vec) => {
+                let iterator = match vec.first() {
+                    Some(Value::Int(i)) => *i,
+                    _ => return Err(not_convertible_error!(v, "expected int")),
+                };
+                let data: &Vec<u8> = match vec.get(1) {
+                    Some(Value::BulkString(d)) => d,
+                    _ => return Err(not_convertible_error!(v, "expected data")),
+                };
+                Ok(DumpResult {
+                    iterator,
+                    data: data.to_vec(),
+                })
+            }
+            _ => Err(not_convertible_error!(v, "expected array")),
+        }
+    }
+}
+
+/// An iterator over the results of the `bf_scan_dump` command.
+///
+/// Examples:
+///
+/// ```rust,no_run
+/// # use std::collections::VecDeque;
+/// use redis::bloom::{AllInfoResponse, DumpIterator, DumpResult};
+/// use redis::{Commands, ToRedisArgs};
+/// # fn do_something() -> redis::RedisResult<()> {
+/// use redis::{PubSubCommands, ControlFlow};
+/// let client = redis::Client::open("redis://127.0.0.1/")?;
+/// let mut con = client.get_connection()?;
+///
+/// let mut chunks: VecDeque<DumpResult> = VecDeque::new();
+///
+/// let dump_iterator = DumpIterator::new(&mut con, "bloomf2");
+/// for chunk in dump_iterator {
+///
+///    chunks.push_back(chunk);
+///  }
+///  println!("completed dump for bloomf2");
+///  Ok(())}
+/// #
+/// ```
+
+pub struct DumpIterator<'a> {
+    con: &'a mut Connection,
+    key: &'a str,
+    iter_id: i64,
+}
+
+impl<'a> DumpIterator<'a> {
+    /// Create a new iterator for the given key
+    pub fn new(con: &'a mut Connection, key: &'a str) -> Self {
+        DumpIterator {
+            con,
+            key,
+            iter_id: 0,
+        }
+    }
+}
+
+impl Iterator for DumpIterator<'_> {
+    type Item = DumpResult;
```

Comment:
it's better that `Item = RedisResult<DumpResult>` - https://github.com/redis-rs/redis-rs/pull/1416